### PR TITLE
F#SRC-5990 Automatic calibration loader not working in URSIM

### PR DESCRIPTION
**Pull request by maxzieba**
## Proposed changes

User story: https://shadowrobot.atlassian.net/browse/SRC-5990
Description: Additional exception handling and fixing the string generation for config file loader.

## Checklist

If the task is applicable to this pull request (see applicability criteria in brackets), make sure it is completed before checking the corresponding box. Otherwise, tick the box right away. Make sure that **ALL** boxes are checked **BEFORE** the PR is merged.

- [X] Manually tested that added code works as intended (if any functional/runnable code is added).
- [X] Added automated tests (if a new class is added (Python or C++), interface of that class must be unit tested).
- [X] Tested on real hardware (if the changed or added code is used with the real hardware).
- [X] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc.)

### CODE REVIEW

File: sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py [hunk 1]
```diff
@@ -89,13 +90,16 @@ def _get_serial_from_arm(self, arm_ip):
             stdin, stdout, stderr = client.exec_command('cat /root/ur-serial')
             arm_serial_number = stdout.readline()
             client.close()
-        except (paramiko.BadHostKeyException, paramiko.AuthenticationException, paramiko.SSHException) as e:
-            ssh_exception_message = "Failed to SSH into arm - {}".format(e.message)
+        except (BadHostKeyException, AuthenticationException, SSHException) as e:
+            ssh_exception_message = " Failed to SSH into arm - {}".format(e.message)
+        except NoValidConnectionsError as e:
+            ssh_exception_message = " Failed to SSH into arm - {}".format(e.errors)
+        except Exception as e:
```
Comment (mikramarc) [inline sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py]: we dont want to handle exceptions that we dont know about

File: sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py [hunk 2]
```diff
@@ -89,13 +90,16 @@ def _get_serial_from_arm(self, arm_ip):
             stdin, stdout, stderr = client.exec_command('cat /root/ur-serial')
             arm_serial_number = stdout.readline()
             client.close()
-        except (paramiko.BadHostKeyException, paramiko.AuthenticationException, paramiko.SSHException) as e:
-            ssh_exception_message = "Failed to SSH into arm - {}".format(e.message)
+        except (BadHostKeyException, AuthenticationException, SSHException) as e:
+            ssh_exception_message = " Failed to SSH into arm - {}".format(e.message)
+        except NoValidConnectionsError as e:
+            ssh_exception_message = " Failed to SSH into arm - {}".format(e.errors)
```
Comment (mikramarc) [inline sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py:96]: why do we do e.errors in one and e.message in another?

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py (hunk 2 above)
Comment (maxzieba) [reply sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py:96]: 'errors' wraps the 'message' attribute
https://www.kite.com/python/docs/paramiko.ssh_exception.NoValidConnectionsError

File: sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py (hunk 1 above)
Comment (maxzieba) [reply sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py]: Changed to more specific 'socket.error'

**Review by giusebar — APPROVED** [review]

**Review by biofotis — APPROVED** [review]

File: sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py [hunk 3]
```diff
@@ -89,13 +90,16 @@ def _get_serial_from_arm(self, arm_ip):
...
+        except socket.error as e:
+            ssh_exception_message = " Failed to SSH into arm - {}.".format(e.message)
```
Comment (mikramarc) [inline sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py]: please merge this with line 93

**Review by mikramarc — CHANGES_REQUESTED** [review]

**Review by mikramarc — APPROVED** [review]

**Pull request merged**